internal/command/machine/run.go

```diff
@@ -383,7 +383,10 @@ func determineImage(ctx context.Context, appName string, imageOrPath string) (im
 			NoCache:    flag.GetBool(ctx, "no-build-cache"),
 		}
 
-		dockerfilePath := cfg.Dockerfile()
+		dockerfilePath := ""
+		if cfg != nil {
+			dockerfilePath = cfg.Dockerfile()
+		}
```

Comment:
what about adding the nil check at:

https://github.com/superfly/flyctl/blob/46ba1f6164672cd3ca67819dd8b976abe68bfbd8/internal/appconfig/config.go#L131

Reply:
`c.Build` wasn't nil there, `c` was.

Reply:
I can if you want, but I really don't like the idea of calling methods on null objects. seems like a bad road to go down imo

Reply:
but the Getter is buggy anyways, we have to fix it otherwise it makes any caller crash

This is common pattern I would say, protobuf generator uses it too, i.e.: [service.pb.go]( https://github.com/superfly/nomad-firecracker/blob/1e6abe700cb0f9bbf99c7d6d14ed4617882b8ebb/internal/flyd/gen/flyd/v1/service.pb.go#L236-L241)

Reply:
Ok, guess it's the rustacean in me being a little overly defensive haha, thanks for the reference :)

Reply:
No doubts, Go sucks in that matter.